pytorch_toolkit/formula_recognition/tools/export.py

```diff
@@ -0,0 +1,324 @@
+                       [1, self.config.get('head', {}).get('decoder_hidden_size', 512)],
+                       [1, self.config.get('head', {}).get('encoder_hidden_size', 256)],
+                       [1, output_h, output_w, self.config.get('head', {}).get('decoder_hidden_size', 512)], [1, 1]]
+        input_shape = "{}, {}, {}, {}, {}".format(*input_shape)
+        export_command = f"""{OPENVINO_DIR}/bin/setupvars.sh &&
+        python {OPENVINO_DIR}/deployment_tools/model_optimizer/mo.py \
+        --framework onnx \
+        --input_model {self.config.get('res_decoder_name')} \
+        --input {self.config.get("decoder_input_names", DECODER_INPUTS)} \
+        --input_shape '{str(input_shape)}' \
+        --output {self.config.get("decoder_output_names", DECODER_OUTPUTS)}"""
+        if self.config.get('verbose_export'):
+            print(export_command)
+        subprocess.run(export_command,
+                       shell=True, check=True
+                       )
+
+
+    def load_ir_model(self):
+        ie = IECore()
+        encoder = read_net(self.config.get("res_encoder_name").replace(".onnx", ".xml"), ie)
+        dec_step = read_net(self.config.get("res_decoder_name").replace(".onnx", ".xml"), ie)
+
+        self.exec_net_encoder = ie.load_network(network=encoder, device_name="CPU")
+        self.exec_net_decoder = ie.load_network(network=dec_step, device_name="CPU")
+
+    def run_ir_model(self, img):
+        enc_res = self.exec_net_encoder.infer(inputs={self.config.get(
+            "encoder_input_names", ENCODER_INPUTS).split(",")[0]: img})
+        enc_out_names = self.config.get("encoder_output_names", ENCODER_OUTPUTS).split(",")
+        ir_row_enc_out = enc_res[enc_out_names[0]]
+        dec_states_h = enc_res[enc_out_names[1]]
+        dec_states_c = enc_res[enc_out_names[2]]
+        output = enc_res[enc_out_names[3]]
+        dec_in_names = self.config.get("decoder_input_names", DECODER_INPUTS).split(",")
+        dec_out_names = self.config.get("decoder_output_names", DECODER_OUTPUTS).split(",")
+        tgt = np.array([[START_TOKEN]] * 1)
+        logits = []
+        for _ in range(self.model.head.max_len):
+            dec_res = self.exec_net_decoder.infer(inputs={
+                dec_in_names[0]: dec_states_h,
+                dec_in_names[1]: dec_states_c,
+                dec_in_names[2]: output,
+                dec_in_names[3]: ir_row_enc_out,
+                dec_in_names[4]: tgt
+            }
+            )
+
+            dec_states_h = dec_res[dec_out_names[0]]
+            dec_states_c = dec_res[dec_out_names[1]]
+            output = dec_res[dec_out_names[2]]
+            logit = dec_res[dec_out_names[3]]
+            logits.append(logit)
+
+            tgt = np.reshape(np.argmax(logit, axis=1), (1, 1)).astype(np.long)
+            if tgt[0][0] == END_TOKEN:
+                break
+
+        return np.argmax(np.array(logits).squeeze(1), axis=1)
+
+    def get_ir_metric(self):
```

Comment:
Looks like `ONNXExported` is not the best place for this method. Do does it related to ONNX Export?

Reply:
Since it tests converted by this class model on one of this class' dataloaders, I think, it is related to this class.

Reply:
Probably it is a kind of `Evaluator`. Since this method performs accuracy metric evaluation but not onnx export.

Reply:
Moved this to a separate Evaluator class